test/rulesets/Base/mapreduce.jl

```diff
@@ -120,6 +120,25 @@ const CFG = ChainRulesTestUtils.ADviaRuleConfig()
         @test rrule(SumRuleConfig(), Base.sum, xs, weights) isa Nothing
     end
 
+    VERSION >= v"1.7" && @testset "maximum(f, xs)" begin
+        test_rrule(maximum, abs, [-4.0, 2.0, 2.0])
+        test_rrule(minimum, sqrt, Float64[1 2; 3 4])
+        test_rrule(maximum, Multiplier(2.0), [2.0, 4.0, 8.0])  # Multiplier defined in test_helpers.jl
+
+        # repeated -- can't use FiniteDifferences
+        y1, bk1 = rrule(CFG, maximum, abs, [-4.0, 2.0, 4.0, 2.0])
+        @test y1 === 4.0
+        @test unthunk(bk1(10.0)[3]) == [-10, 0, 0, 0]
+
+        # dims keyword -- these need to call `rrule_via_ad(broadcast, ...`, which needs AD
+        @test_skip test_rrule(maximum, sqrt, Float64[1 2; 3 4], fkwargs=(; dims = 1), check_inferred=false)
+        @test_skip test_rrule(minimum, abs, randn(3,3), fkwargs=(; dims = 2), check_inferred=false)
```

Comment:
these will need https://github.com/JuliaDiff/FiniteDifferences.jl/pull/203

Reply:
I thought these needed https://github.com/JuliaDiff/ChainRulesTestUtils.jl/pull/243 : with `dims` it always calls `broadcast`.

Reply:
But where do InplaceableThunks come from? This path of this rule doesn't make them. 

I do still get an error with only CRTU update:
```
julia> test_rrule(maximum, sqrt, Float64[1 2; 3 4], fkwargs=(; dims = 1), check_inferred=false)
test_rrule: maximum on typeof(sqrt),Matrix{Float64}: Error During Test at /Users/me/.julia/packages/ChainRulesTestUtils/fCvaU/src/testers.jl:193
  Got exception outside of a @test
  DimensionMismatch("second dimension of A, 4, does not match length of x, 7")
  Stacktrace:
    [1] gemv!(y::Vector{Float64}, tA::Char, A::Matrix{Float64}, x::Vector{Float64}, α::Bool, β::Bool)
      @ LinearAlgebra ~/.julia/dev/julia/usr/share/julia/stdlib/v1.8/LinearAlgebra/src/matmul.jl:493
    [2] mul!
      @ ~/.julia/dev/julia/usr/share/julia/stdlib/v1.8/LinearAlgebra/src/matmul.jl:93 [inlined]
    [3] mul!
      @ ~/.julia/dev/julia/usr/share/julia/stdlib/v1.8/LinearAlgebra/src/matmul.jl:276 [inlined]
    [4] *(tA::Transpose{Float64, Matrix{Float64}}, x::Vector{Float64})
      @ LinearAlgebra ~/.julia/dev/julia/usr/share/julia/stdlib/v1.8/LinearAlgebra/src/matmul.jl:86
    [5] _j′vp(fdm::FiniteDifferences.AdaptedFiniteDifferenceMethod{5, 1, FiniteDifferences.UnadaptedFiniteDifferenceMethod{7, 5}}, f::Function, ȳ::Vector{Float64}, x::Vector{Float64})
      @ FiniteDifferences ~/.julia/packages/FiniteDifferences/R6uao/src/grad.jl:80
    [6] j′vp(fdm::FiniteDifferences.AdaptedFiniteDifferenceMethod{5, 1, FiniteDifferences.UnadaptedFiniteDifferenceMethod{7, 5}}, f::ChainRulesTestUtils.var"#fnew#45"{ChainRulesTestUtils.var"#call#41"{Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}}, Tuple{typeof(broadcast), typeof(sqrt), Matrix{Float64}}, Tuple{Bool, Bool, Bool}}, ȳ::InplaceableThunk{Thunk{ChainRules.var"#1316#1319"{Matrix{Float64}, Int64, Matrix{Float64}, ProjectTo{AbstractArray, NamedTuple{(:element, :axes), Tuple{ProjectTo{Float64, NamedTuple{(), Tuple{}}}, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}}}, Matrix{CartesianIndex{2}}}}, ChainRules.var"#1317#1320"{Matrix{Float64}, Int64, Matrix{CartesianIndex{2}}}}, x::Matrix{Float64})
      @ FiniteDifferences ~/.julia/packages/FiniteDifferences/R6uao/src/grad.jl:73
    [7] _make_j′vp_call(fdm::Any, f::Any, ȳ::Any, xs::Any, ignores::Any)
      @ ChainRulesTestUtils ~/.julia/packages/ChainRulesTestUtils/fCvaU/src/finite_difference_calls.jl:51
    [8] f_pb
      @ ~/.julia/packages/ChainRulesTestUtils/fCvaU/src/rule_config.jl:40 [inlined]
    [9] (::ChainRules.var"#minormax_f_back2#2098"{ChainRules.var"#maximum_pullback#1326"{ChainRules.var"#findmax_pullback#1318"{Int64
    ```

Reply:
Solved by the to_vec PR, as you said.

Can this thing give less cryptic errors than these "DimensionMismatch" when it goes wrong?

Reply:
Yeah, I agree with you in general: https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/244

Here though this is coming from `rrule_via_ad` using the `make_v'jp_call` rather than the usual place 😂 

Solving https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/213 would be a big QoL improvement indeed. It's on my list

Reply:
Great! 

This one is weird locally, but on 1.6 it seems to work (or will once changed to ` ≈ [10 0 0; 0 -20 0]`):
```
julia> y2, bk2 = rrule(CFG, minimum, abs, [1 2 3; -5 -4 -4], dims = 2);

julia> @test y2 == hcat([1, 4])
Test Passed
  Expression: y2 == hcat([1, 4])
   Evaluated: [1; 4;;] == [1; 4;;]

julia> bk2(hcat([10, 20]))
(NoTangent(), NoTangent(), NoTangent())
```